### PR TITLE
ci: add auto-submit workflow for snapshot/timestamp

### LINKED ISSUE #553: Automate merging of snapshot/timestamp PR
**Description**

Previously we didn't automate this due to lack of e2e tests and a lack of a staging environment. Now that we have good testing, probes, a preprod environment, and issue filing for failed automation, I'd like to revisit this. Merging the snapshot/timestamp PR has been a source for issues in the past, especially over holidays. Since we have a long break coming up, I think we should prioritize this automation.

cc @asraa @kommendorkapten for thoughts

### PR DESCRIPTION
**Pull request by asraa**
Signed-off-by: Asra Ali <asraa@google.com>

This PR adds an auto-submit bot for the snapshot/timestamp updates. The workflow will still trigger, and then this will run every 6 hours to (1) Check for any outstanding snapshot/timestamp updates (2) Start a review and approve and (3) Attempt to merge.

Fixes https://github.com/sigstore/root-signing/issues/553

cc @haydentherapper

### CODE REVIEW

File: .github/workflows/review-snapshot-timestamp.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  workflow_dispatch:
+
+jobs:
+  review:
+    runs-on: ubuntu-20.04
+    permissions:
+      pull-requests: 'write'
+      contents: 'write'
+    env:
+      GITHUB_TOKEN: ${{ secrets.SIGSTORE_ROOT_SIGNING_FINE_GRAINED_PAT }}
```
Comment (asraa) [inline .github/workflows/review-snapshot-timestamp.yml:36]: @cpanato Carlos, quick ask -- when you have some time could you replace this with a sigstore-bot that has pull request and content write permissions?

Comment (cpanato) [reply .github/workflows/review-snapshot-timestamp.yml:36]: sure thing

File: .github/workflows/scripts/review-pull-request.sh [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    -f event='APPROVE'
+
+    # Attempt to merge PR
+    GH_TOKEN="${GITHUB_TOKEN}" gh api \
+    --method PUT \
+    -H "Accept: application/vnd.github+json" \
+    /repos/"${GITHUB_REPOSITORY}"/pulls/"${PULL_NUMBER}"/merge \
+    -f commit_title='Update Snapshot and Timestamp' \
+    -f commit_message='update snapshot and timestamp' 
+fi
```
Comment (cpanato) [inline .github/workflows/scripts/review-pull-request.sh:50]: nit: new line 😬

File: .github/workflows/review-snapshot-timestamp.yml (hunk 1 above)
Comment (cpanato) [reply .github/workflows/review-snapshot-timestamp.yml:36]: done

File: .github/workflows/scripts/review-pull-request.sh [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+timestamp_update() {
+    gh api -H "Accept: application/vnd.github.v3+json" "/repos/${GITHUB_REPOSITORY}/pulls?head=sigstore:update-snapshot-timestamp" | jq '.[0]' | jq 'select (.!=null)'
+}
+
+UPDATE_PR=$(timestamp_update)
+if [[ -z "${UPDATE_PR}" ]]; then
+    PULL_NUMBER=$(echo "${UPDATE_PR}" | jq -r '.number')
+
+    # Approve PR
+    curl \
```
Comment (cpanato) [inline .github/workflows/scripts/review-pull-request.sh:29]: you can use the `gh pr review PR_NUMBER --approve` i guess

**Review by cpanato — APPROVED** [review]

**Comment by cpanato** [comment]
i wll merge and see if it will work, if not we can follow up on that

**Comment by cpanato** [comment]
i will trigger that manually instead of wait for 6 hours :)

**Comment by cpanato** [comment]
https://github.com/sigstore/root-signing/actions/runs/3600760731/jobs/6065777450

**Comment by cpanato** [comment]
`./.github/workflows/scripts/report-failure.sh: line 45: ISSUE_REPOSITORY: unbound variable`

**Pull request merged**